src/elements/content-sidebar/activity-feed/activity-feed/__tests__/ActivityThread.test.js

```diff
@@ -35,16 +39,28 @@ describe('src/elements/content-sidebar/activity-feed/activity-feed/ActivityThrea
         expect(getByText('Test')).toBeVisible();
     });
 
-    test('should call onShowReplies on button click', () => {
+    test('should call onShowReplies on button click if total replies is greater than the number of replies in Feed', () => {
         const onShowReplies = jest.fn();
-        const { getByText } = getWrapper({ onShowReplies });
+        const replies = [cloneDeep(repliesMock[0])];
+        const { getByText } = getWrapper({ onShowReplies, replies });
 
         const button = getByText(localize(messages.showReplies.id, { repliesToLoadCount: 1 }));
         expect(button).toBeVisible();
         fireEvent.click(button);
 
         expect(onShowReplies).toBeCalled();
-        expect(getByText(localize(messages.hideReplies.id))).toBeVisible();
+    });
+
+    test('should call onHideReplies on button click if total replies is equal to the number of replies in Feed', () => {
+        const onHideReplies = jest.fn();
+        const lastReply = cloneDeep(repliesMock[repliesMock.length - 1]);
+        const { getByText } = getWrapper({ onHideReplies });
+
+        const button = getByText(localize(messages.hideReplies.id));
+        expect(button).toBeVisible();
+        fireEvent.click(button);
+
+        expect(onHideReplies).toBeCalledWith(lastReply);
     });
```

Comment:
Lets add some edge cases mentioned above where the total is 0 or the number of replies is 0 and vise versa

Reply:
I've modified the test for not rendering button with multiple cases. This test actually covers the rendering of button and calling `onHideReplies` on click. I've also updated the description so that it's clearer.